### ISSUE TITLE
Pulseeffects crashing all the time while enabling applications

### ISSUE THREAD

**Issue by rory4ever** [wwmm/easyeffects#597]
Hi,

After installing Pulseeffects everything works as it should but after a reboot Pulseeffects starts normally but when an application (for example VLC player or Firefox (youtube clip) is playing, Pulseeffects just disappears/exits and I cannot use it anymore. 

I have the following configuration / setup: 
- Ubuntu 18.04.3 LTS - fully updated
- Pulseeffects 4.6.8.  - Pulseaudio 12.2
- soundcard: Cambridge Dacmagic PLUS DAC, my internal on-board soundcard has been disabled
  the soundcard works great via ALSA. 

Short reproduction path: 
- after a fresh reboot, start Pulseeffects
- Pulseeffects window shows up normally. 
- Make sure to leave the default soundcard as it is now (sound settings in ubuntu)
- Start browser, go to Youtube or any other media site, click on any Youtube vid but don't play it yet
- In Pulseeffects I now see my browser at " applications"  which is good, it's turned "on" 
- In the browser, start hte Youtube video. 
Expected result: video plays, Pulseeffects window stays open and I can use it normally
Actual result: after 1-2 seconds of starting the video in my browser, pulseeffects exits
I hear no sound. I have to reselect my default sound card in Ubuntu -> sound settings to get sound again. My video just played along during all this. 

I installed Pavuconrol and that is working fine. 

I am not an experienced Linux user so you probably need more info from me. Please tell me what info you need (i.e. where to find the log file?) to solve this issue. Thanks in advance!

Best regards,
Roy

**Comment by wwmm** [comment]
It would be good to know how you installed PulseEffects. Was it thought flatpak or through a native debian package? In any case there is some unknown problem happening in ubuntu 18 as you can see here #551. 

Kill any running PE instance through the command `pulseeffects -q` and then start it in debug mode `G_MESSAGES_DEBUG=pulseeffects pulseeffects`. It will print some lines in the terminal.

**Comment by rory4ever** [comment]
Thanks for replying. I have collected all the info and logs the " how to report bugs" page is suggesting and attached it to this thread. It's quite a lot of data. 
[pulseeffects_LOGS.txt](https://github.com/wwmm/pulseeffects/files/3991322/pulseeffects_LOGS.txt)
[pesettings.txt](https://github.com/wwmm/pulseeffects/files/3991323/pesettings.txt)
![Screenshot from 2019-12-21 12-10-43](https://user-images.githubusercontent.com/39132289/71307272-2ba97000-23ec-11ea-9f44-0edc7a5615d3.png)
![Screenshot from 2019-12-21 12-10-40](https://user-images.githubusercontent.com/39132289/71307273-2ba97000-23ec-11ea-9cac-ea6d846c6cb6.png)
![Screenshot from 2019-12-21 12-10-39](https://user-images.githubusercontent.com/39132289/71307274-2c420680-23ec-11ea-8d51-5200fbfec744.png)
![Screenshot from 2019-12-21 12-10-36](https://user-images.githubusercontent.com/39132289/71307275-2c420680-23ec-11ea-8549-a7b60f39a1c8.png)
![Screenshot from 2019-12-21 12-10-33](https://user-images.githubusercontent.com/39132289/71307276-2c420680-23ec-11ea-8f78-13a508e18f58.png)

**Comment by wwmm** [comment]
Your problem is a little different. Based on PE logs it seems our playback pipeline is being killed when it starts to play. Run the command `sudo coredumpctl info` right after PulseEffects crashes. It may give us some information about why we are being killed.

**Comment by wwmm** [comment]
I saw in your config that the pitch plugin is the only one enabled. Try to disable it before starting an audio application. Or reset Pulseeffects `pulseeffects -r`. For some unknown reason this plugin is crashing in some systems. I am almost considering removing it.

**Comment by rory4ever** [comment]
Thanks again. 

So after it crashes (exits) I typed sudo coredumpctl info
However, the result was: 
No coredumps found.

I think it didn't really "crash" ? 

So when disabling the pitch plugin pulseffects like you suggested, it keeps on running fine and doesn't crash. As soon as I enable toe pitch plugin, the sounds stops immediately and 
pulseeffects exits. 

So I reset and restarted PE:
when I select any other plugin it works. 
I then moved "pitch" plugin to the top (it was at the bottom) and tried it again, now it suddenly works!
however, when I exit PE and restart it, it crashes again (some initialisation problem or something like that?)

when PE is running and I select the pitch shifter plugin I see the following in the terminal:

RubberBandPitchShifter::runImpl: buffer underrun: required = 512, available = 0
RubberBandPitchShifter::runImpl: buffer underrun: required = 512, available = 212

When I disable this plugin before I exit PE, PE starts normally and i can select the pitch plugin and it works OK. 
Tried this 5 times or so so the behavior is stable and as expected. 

So the problem occurs only when the pitch shifter plugin is enabled and I exit PE and PE remembers that this plugin is enabled and then when I start PE and play something it crashes. 

It IS workable for me this way, one of the main reasons to run PE is because of the excellent Rubberband plugin and i don't want a sophisticated way using Jack configs.

**Comment by wwmm** [comment]
Interesting. When I have time I will try to do some tests with this plugin again and see why it is behaving this way.

**Comment by Orochimarufan** [comment]
After a system update PulseEffects receives a SIGKILL when any new stream starts. It seems to happen if Crystalizer is the first or only effect in the pipeline. Also, it works fine when opening PE or enabling the effect with a stream is already running. I didn't manage to get any meaningful diagnostics information, but here are the last debug lines before the crash:
```
(pulseeffects:2496): pulseeffects-DEBUG: 02:51:26.214: sie: using output device: alsa_output.pci-0000_00_1b.0.analog-stereo
(pulseeffects:2496): pulseeffects-DEBUG: 02:51:26.214: pulse_settings_ui: output device changed: alsa_output.pci-0000_00_1b.0.analog-stereo
(pulseeffects:2496): pulseeffects-DEBUG: 02:51:26.214: pulse_settings_ui: added sink: alsa_output.pci-0000_00_1b.0.analog-stereo
(pulseeffects:2496): pulseeffects-DEBUG: 02:51:26.215: pulse_settings_ui: added source: alsa_output.pci-0000_00_1b.0.analog-stereo.monitor
(pulseeffects:2496): pulseeffects-DEBUG: 02:51:26.216: soe: using input device: alsa_input.pci-0000_00_1b.0.analog-stereo
(pulseeffects:2496): pulseeffects-DEBUG: 02:51:26.216: pulse_settings_ui: input device changed: alsa_input.pci-0000_00_1b.0.analog-stereo
(pulseeffects:2496): pulseeffects-DEBUG: 02:51:26.216: pulse_settings_ui: added source: alsa_input.pci-0000_00_1b.0.analog-stereo
(pulseeffects:2496): pulseeffects-DEBUG: 02:51:26.226: sie: rtkit: changed GstPulseSrc:source thread real-time priority value to 6
(pulseeffects:2496): pulseeffects-DEBUG: 02:51:26.247: sie: NULL -> READY -> PLAYING
(pulseeffects:2496): pulseeffects-DEBUG: 02:51:26.250: sie: READY -> PAUSED -> PLAYING
(pulseeffects:2496): pulseeffects-DEBUG: 02:51:26.267: sie: rtkit: changed GstPulseSrc:source thread real-time priority value to 6
(pulseeffects:2496): pulseeffects-DEBUG: 02:51:26.268: sie: sampling rate: 48000 Hz
(pulseeffects:2496): pulseeffects-DEBUG: 02:51:26.352: sie: pulsesrc latency [us]: 1000
(pulseeffects:2496): pulseeffects-DEBUG: 02:51:26.352: sie: pulsesrc buffer [us]: 200000
(pulseeffects:2496): pulseeffects-DEBUG: 02:51:26.352: sie: total latency: 0 ms
(pulseeffects:2496): pulseeffects-DEBUG: 02:51:26.376: peadapter: pulseaudio block size 48 frames
(pulseeffects:2496): pulseeffects-DEBUG: 02:51:26.376: peadapter: we will read in chunks of 256 frames
(pulseeffects:2496): pulseeffects-DEBUG: 02:51:26.392: sie: pulsesink latency [us]: 1000
(pulseeffects:2496): pulseeffects-DEBUG: 02:51:26.393: sie: pulsesink buffer [us]: 200000
(pulseeffects:2496): pulseeffects-DEBUG: 02:51:26.393: sie: total latency: 0 ms
(pulseeffects:2496): pulseeffects-DEBUG: 02:51:26.520: sie: total latency: 126 ms
(pulseeffects:2496): pulseeffects-DEBUG: 02:51:26.589: sie: total latency: 126 ms
(pulseeffects:2496): pulseeffects-DEBUG: 02:51:26.591: sie: PAUSED -> PLAYING -> VOID_PENDING
(pulseeffects:2496): pulseeffects-DEBUG: 02:51:26.592: sie: total latency: 126 ms
[1]    2496 killed     G_MESSAGES_DEBUG=pulseeffects pulseeffects
```

The relevant package versions (on Arch) are:
```
23392 [2020-04-07T02:43:33+0200] [ALPM] upgraded ffmpeg (1:4.2.2-4 -> 1:4.2.2-5)
23414 [2020-04-07T02:43:37+0200] [ALPM] upgraded gst-plugins-bad-libs (1.16.2-5 -> 1.16.2-7)
23422 [2020-04-07T02:43:38+0200] [ALPM] upgraded gst-plugins-bad (1.16.2-5 -> 1.16.2-7)
23439 [2020-04-07T02:44:06+0200] [ALPM] upgraded lsp-plugins (1.1.15-1 -> 1.1.17-1)
```

It seemed reproducible for a while but unfortunately I don't seem to be able to reproduce it anymore after down- and re-upgrading ffmpeg. This is especially strange because 1:4.2.2-5 was only a rebuild due to a new x265 release, so I don't see how it should have affected anything.

I was going to open a new issue, but seeing as I can't reproduce it anymore and the symptoms seem similar to this one, I opted to post it here instead. I'll report back should it happen again.

**Comment by wwmm** [comment]
I use Arch too but somehow I never saw that. Thee plugin with which I have andom problems is the pitch plugin.

**Comment by wwmm** [comment]
As EasyEffects 6.0.0 does not use GStreamer anymore it will probably be better to close this issue and open a new one if such problems are also present with native Pipewire filters.
